src/main/java/de/tum/cit/aet/artemis/core/web/admin/AdminBuildJobQueueResource.java

```diff
@@ -95,8 +95,8 @@ public ResponseEntity<List<BuildAgentInformation>> getBuildAgentSummary() {
     @GetMapping("build-agent")
     public ResponseEntity<BuildAgentInformation> getBuildAgentDetails(@RequestParam String agentName) {
         log.debug("REST request to get information on build agent {}", agentName);
-        BuildAgentInformation buildAgentDetails = localCIBuildJobQueueService.getBuildAgentInformation().stream().filter(agent -> agent.name().equals(agentName)).findFirst()
-                .orElse(null);
+        BuildAgentInformation buildAgentDetails = localCIBuildJobQueueService.getBuildAgentInformation().stream().filter(agent -> agent.buildAgent().name().equals(agentName))
+                .findFirst().orElse(null);
         return ResponseEntity.ok(buildAgentDetails);
```

Comment:
I know you didn't add this code, but maybe we can still discuss its validity :) As I see it here, we'd return `ResponseEntity.ok(null)` when the respective agent is not found, right? This seems like wrapping an error value in a success code